Update sample with basemap gallery toolkit component (#519)
Esri/arcgis-maps-sdk-flutter-samples: lib/samples/set_basemap/README.md

## How to use the sample

> Tap the map button to show or hide the basemap gallery. Select an item in the gallery to apply it to the map.

Reviewer: ```suggestion
Tap the map button to show or hide the basemap gallery. Select a basemap in the gallery to apply it to the map.
```

## How it works

1. Create an `ArcGISMap` object with an initial basemap style.
2. Set the map to the `ArcGISMapViewController` object.

> 3. Create a `BasemapGalleryController` and set its `geoModel` to the map.

Reviewer: ```suggestion
3. Create a `BasemapGalleryController` and set the map to the `geoModel` property.
```

> 4. Show the `BasemapGallery` toolkit component and let users select a basemap.

Reviewer: ```suggestion
4. Create a `BasemapGallery` widget using the `BasemapGalleryController`.
```
